### ISSUE TITLE
`function-max-empty-lines` writes a nested call a second time over the code behind the outer one

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#366]
### What minimal example or steps are needed to reproduce the bug?

`function-max-empty-lines` rewrites a nested call twice, and the second copy is written over the code standing behind the outer one. No comment and no custom syntax are needed — plain CSS reaches it.

Linted with `--fix` under the option `0`:

| code | what comes back |
|---|---|
| `a { b: f(`⏎⏎⏎`g(`⏎⏎⏎`1)); }` | `a { b: f(`⏎`g(`⏎`1))g(`⏎`1)); }` |

Two warnings are reported, one for each call, and both are right: each call really does hold more empty lines than the option allows. What the fix writes is not.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/function-max-empty-lines": 0
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

Every run of empty lines collapsed to what the option allows, wherever it stands, and the value otherwise left alone.

### What actually happened?

The inner call is written a second time, after the outer one has already carried it, and everything the outer call held behind the inner one is duplicated with it.

### What is the cause?

The fix does not rewrite the value in place. It collects pairs of `[the text in front of this node, this node rewritten]` into `splittedValue` and moves `sourceIndexStart` to `node.sourceIndex + stringifiedNode.length`, then joins the pairs and appends `stringValue.slice(sourceIndexStart)` once the walk is over.

`postcss-value-parser` walks a parent before its children, and the outer call's `stringifiedNode` already holds the inner one, rewritten along with the rest of it. So when the inner call is reached, `sourceIndexStart` already stands behind it: `stringValue.slice(sourceIndexStart, node.sourceIndex)` is asked for a slice that runs backwards and hands back an empty string, and the inner call's own rewritten text is appended to what the outer call has already written.

### Do you have a proposal to fix the bug?

A node whose `sourceIndex` is behind `sourceIndexStart` has been rewritten already, by an ancestor that carried it, and there is nothing left for it to write. The walk can decline to descend into a call it has just fixed — `return false` from the callback where the fix ran — or the fix can pass over a node standing behind the position it has reached. The first says what is meant: the outer call has answered for everything inside it.

The warning is a second question. Two warnings for one run of empty lines counted twice over is arguably right, since each call does hold them; whether the inner one is worth reporting after the outer one has been fixed is a matter of what the rule wants to say, and either answer leaves the value correct.
